File: 卡码/代码随想录/1/08-贪心算法/C++/03-困难/LeetCode_0056.cpp
//56. 合并区间
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
	vector<vector<int>> merge(vector<vector<int>>& intervals) {
		vector<vector<int>> res;
		sort(intervals.begin(),intervals.end(),[](vector<int>& a,vector<int>& b){
			return a[0]<b[0];
		});
		res.push_back(intervals[0]);
		for(int i=1;i<intervals.size();i++){
			if(intervals[i][0]<=res.back()[1]){
				res.back()[1]=max(res.back()[1],intervals[i][1]);
			}else{
				res.push_back(intervals[i]);
			}
		}
		return res;
	}
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/1/08-贪心算法/Java/greedy/src/main/java/com/kurumi/greedy/_03_困难/LeetCode_0056.java
//56. 合并区间
package com.kurumi.greedy._03_困难;

import java.util.Arrays;
import java.util.LinkedList;

public class LeetCode_0056 {

}
class Solution {
  public int[][] merge(int[][] intervals) {
    LinkedList<int[]> res=new LinkedList<>();
    Arrays.sort(intervals,(a,b)->Integer.compare(a[0], b[0]));
    res.add(intervals[0]);
    for(int i=1;i<intervals.length;i++){
      if(intervals[i][0]<=res.getLast()[1]){
        res.getLast()[1]=Integer.max(intervals[i][1], res.getLast()[1]);
      }else{
        res.add(intervals[i]);
      }
    }
    return res.toArray(new int[res.size()][]);
  }
}
